### ISSUE TITLE
bars and punctum mora

### ISSUE THREAD

**Issue by eroux** [gregorio-project/gregorio#795]
This is a first step to #767 

Gregorio used to do that so I'm labelling this issue "bug", although it might have been removed on purpose. The idea is to align bars in the middle of notes, not taking into account the punctum mora.

Current gregorio behavior for

```
Ió(g.) (,) i(g) (,) i(g)
```

is

![mora-bar](https://cloud.githubusercontent.com/assets/60868/12397598/84b37d46-be0e-11e5-89f6-88d0fa66aa0c.png)

while the first bar should be moved left from (with of punctum mora + width between punctum mora and note).

**Comment by rpspringuel** [comment]
Do you have any idea when the feature may have been removed?

**Comment by eroux** [comment]
I'd say between one and two years ago... but I'm not really sure... 

The story is quite funny : I did that to be in line with old Solesmes books, but then Solesmes changed in their recent books, not really on purpose (their typesetting system was quite poor), and everyone wanted to have "the new Solesmes way" (so the feature was removed from Gregorio), but now Solesmes realized it was a mistake and goes back to the old way...

**Comment by eroux** [comment]
IIRC, there was a TeX variable set to true if the last note sequence ends with a punctum mora, and before typesetting a bar (with no text, not the others), there was a kern of `- widht of punctum mora - with between note and punctum mora`. It should be quite easy to do, or maybe it's just commented, but I admit I easily get lost in the code now... (and I really have to hurry on other Gregorio features)

**Comment by rpspringuel** [comment]
So probably end of 2013 or anytime during 2014.  Let me do some hunting and see if I can find the change.

**Comment by rpspringuel** [comment]
Well, after looking through the commits in 2014, 2013, 2011, and the last few months of 2010, I can't identify where the change was made.  At this point it's probably easier to just start from scratch.

Should this change be implemented as an option, or simply a change in behavior?

**Comment by rpspringuel** [comment]
Wait a minute...  Found the commit where the centering was actually implemented: f771465.  That doesn't answer the question about when (or why) it was removed, but it does confirm @eroux's memory of how the feature worked.

**Comment by henryso** [comment]
Since this has been like this for (more than) five years, I think it would make sense to make this an option that can be toggled between scores for those people who happen to like it better.

**Comment by rpspringuel** [comment]
The previous commit is where the original centering feature was introduced, not when it was removed.  Having found the original implementation though, I've been able to identify that the shifting code is still in the `gregoriotex-signs.tex`.  It was placed inside a `\ifnum#2=1` statement by @eroux in e5cc696 (April 2015).  It seems the intent was to prevent this shifting when inside a bar syllable (a bar with text underneath?).  Presumably then, parameter `#2` should be 1 when there is no text below the bar symbol.  The accompanying comments, however, indicate that the parameter isn't used currently.  This makes me think that the change was started but not finished.

Doing a grep on the current code for `\gre@writebar{[0-9]}{1}` I find that there are 10 instances where `\gre@writebar` is called with `#2` is 1.  Searching for `\gre@writebar{[0-9]}{[^1]}` reveals a symmetrical 10 instances where `#2` is 0.

I'm out of time for today, but the next step is to figure out if @eroux's example is instantiating one of the first group (in which case the problem must be in the shift itself) or one of the second (in which case it's identifying itself wrong somehow.
